evm/logic/context.py

```diff
@@ -193,3 +195,30 @@ def returndatacopy(computation):
 
 def sighash(computation):
     computation.stack.push(computation.msg.sig_hash)
+
+
+def paygas(computation):
+    gas_price = computation.stack.pop(type_hint=constants.UINT256)
+
+    # Only valid if (1) triggered in a top level call and
+    # (2) not been set already during this transaction execution
+    if computation.msg.depth == 0:
+        try:
+            computation.msg.paygas_gasprice(gas_price)
```

Comment:
This looks like an error.  You are calling a property which I believe should raise an exception stating that either `NoneType` or the integers are not callable.  

I do however like using a setter function rather than assigning to a property that throws an exception.  Maybe rename to something like `computation.msg.set_PAYGAS_gasprice(...)`?

Reply:
My mistake here. I will update the setter to `set_PAYGAS_gasprice` function.

Reply:
Do you think it's worth implementing a few minimal test cases for this opcode to hit these code paths?  I know the JSON test suite will cover these eventually, but

- we'd probably benefit from having basic test cases for opcodes that were not buried in the hard to decipher JSON fixture tests.
- until we have JSON fixture tests for this it'd be nice to know it works (from a testing/CI perspective)